test/Microsoft.ML.CodeAnalyzer.Tests/TypeIsSchemaShapeTest.cs

```diff
@@ -2,94 +2,89 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Testing;
 using Microsoft.ML.CodeAnalyzer.Tests.Helpers;
 using Xunit;
+using VerifyCS = Microsoft.ML.CodeAnalyzer.Tests.Helpers.CSharpCodeFixVerifier<
+    Microsoft.ML.Analyzer.TypeIsSchemaShapeAnalyzer,
+    Microsoft.CodeAnalysis.Testing.EmptyCodeFixProvider>;
 
 namespace Microsoft.ML.Analyzer.Tests
 {
-    public sealed class TypeIsSchemaShapeTest : DiagnosticVerifier<TypeIsSchemaShapeAnalyzer>
+    public sealed class TypeIsSchemaShapeTest
     {
         private static string _srcResource;
         internal static string Source => TestUtils.EnsureSourceLoaded(ref _srcResource, "TypeIsSchemaShapeResource.cs");
 
         [Fact]
-        public void ReturnTypeIsSchemaShape()
+        public async Task ReturnTypeIsSchemaShape()
         {
-            var analyzer = GetCSharpDiagnosticAnalyzer();
-            var diag = analyzer.SupportedDiagnostics[0];
-            var diagC = analyzer.SupportedDiagnostics[2];
-
-            string p(string i = "") => string.IsNullOrEmpty(i) ? "" : $" of item {i}";
-
             var expected = new DiagnosticResult[] {
-                diag.CreateDiagnosticResult(23, 13, p()),
-                diag.CreateDiagnosticResult(24, 13, p("bad")),
-                diag.CreateDiagnosticResult(30, 13, p("c.Item2")),
-                diag.CreateDiagnosticResult(40, 13, p("listen")),
+                VerifyCS.Diagnostic(TypeIsSchemaShapeAnalyzer.ShapeDiagnostic.Rule).WithLocation(23, 13).WithArguments(""),
+                VerifyCS.Diagnostic(TypeIsSchemaShapeAnalyzer.ShapeDiagnostic.Rule).WithLocation(24, 13).WithArguments(" of item bad"),
+                VerifyCS.Diagnostic(TypeIsSchemaShapeAnalyzer.ShapeDiagnostic.Rule).WithLocation(30, 13).WithArguments(" of item c.Item2"),
+                VerifyCS.Diagnostic(TypeIsSchemaShapeAnalyzer.ShapeDiagnostic.Rule).WithLocation(40, 13).WithArguments(" of item listen"),
             };
 
-            VerifyCSharpDiagnostic(Source, expected);
+            var test = new VerifyCS.Test { TestCode = Source };
+            test.ExpectedDiagnostics.AddRange(expected);
+            test.Exclusions &= ~AnalysisExclusions.GeneratedCode;
```

Comment:
📝 This means the analyzer reports diagnostics even in code marked as generated (most commonly due to `// <auto-generated/>` in the file header, but also handles certain designer files, etc.). If this was not intentional, we should update the analyzer to indicate these files should be excluded from diagnostic reporting.

Some other analyzers in this test suite require this as well. Please review all of them to ensure exclusions are applied/not applied where you want.